core/trino-main/src/main/java/io/trino/SystemSessionProperties.java

```diff
@@ -531,11 +534,15 @@ public SystemSessionProperties(
                         false,
                         value -> validateIntegerValue(value, MAX_RECURSION_DEPTH, 1, false),
                         object -> object),
-                booleanProperty(
+                new PropertyMetadata<>(
                         USE_MARK_DISTINCT,
                         "Implement DISTINCT aggregations using MarkDistinct",
-                        optimizerConfig.isUseMarkDistinct(),
-                        false),
+                        createUnboundedVarcharType(),
+                        UseMarkDistinct.class,
+                        optimizerConfig.getUseMarkDistinct(),
+                        false,
+                        value -> UseMarkDistinct.fromPropertyValue((String) value),
```

Comment:
Why is this needed? See how `JOIN_DISTRIBUTION_TYPE` is handled

Reply:
> this was to support old values (true and false). Will drop it in place of new property

I don't think it does. `set session x=true` is different than `set session x='true'`